packages/app-desktop/gui/NoteListWrapper/NoteListWrapper.tsx

```diff
@@ -40,30 +40,29 @@ const StyledRoot = styled.div`
 	width: 100%;
 `;
 
-const getTextWidth = (newNoteRef: React.MutableRefObject<any>, text: string): number => {
+const getTextWidth = (newNoteRef: Element, text: string): number => {
 	const canvas = document.createElement('canvas');
 	if (!canvas) throw new Error('Failed to create canvas element');
 	const ctx = canvas.getContext('2d');
 	if (!ctx) throw new Error('Failed to get context');
-	const fontWeight = getComputedStyle(newNoteRef.current).getPropertyValue('font-weight');
-	const fontSize = getComputedStyle(newNoteRef.current).getPropertyValue('font-size');
-	const fontFamily = getComputedStyle(newNoteRef.current).getPropertyValue('font-family');
+	const fontWeight = getComputedStyle(newNoteRef).getPropertyValue('font-weight');
+	const fontSize = getComputedStyle(newNoteRef).getPropertyValue('font-size');
+	const fontFamily = getComputedStyle(newNoteRef).getPropertyValue('font-family');
 	ctx.font = `${fontWeight} ${fontSize} ${fontFamily}`;
 	return ctx.measureText(text).width;
 };
 
 // Even though these calculations mostly concern the NoteListControls component, we do them here
 // because we need to know the height of that control to calculate the note list height.
-const useNoteListControlsBreakpoints = (width: number, newNoteRef: React.MutableRefObject<any>, selectedFolderId: string) => {
+const useNoteListControlsBreakpoints = (width: number, newNoteRef: Element, selectedFolderId: string) => {
 	const [dynamicBreakpoints, setDynamicBreakpoints] = useState<Breakpoints>({ Sm: BaseBreakpoint.Sm, Md: BaseBreakpoint.Md, Lg: BaseBreakpoint.Lg, Xl: BaseBreakpoint.Xl });
 	const previousWidth = usePrevious(width);
 	const widthHasChanged = width !== previousWidth;
 	const showNewNoteButton = selectedFolderId !== getTrashFolderId();
 
 	// Initialize language-specific breakpoints
 	useEffect(() => {
-		if (!widthHasChanged) return;
```

Comment:
Strictly speaking this check is correct and should not be removed. If the width hasn't changed, there's no reason to recalculate the breakpoints. But as mentioned maybe there's something else missing here.

Reply:
The case when width hasn't changed but need breakpoint calculations: when startup with Trash can selected (new note button hidden), then click on a note (new note button shown). No panel width changed but need to calculate breakpoint, or else the button will be big and overflow panel

Reply:
> (new note button hidden), then click on a note (new note button shown).

So if this effect has a dependency on the button being hidden or shown, can this info be passed to it? As a result the effect should trigger again whenever the button visibility changes.

Reply:
Yes, the showNewNoteButton is a useEffect dependency already in previous versions. Yet, the problem is still useRef above, if useRef is null, even effect trigger, it cant update.

Reply:
Ok, fair enough, I feel there's something off in the implementation but the existing code wasn't great either so that might be related.

At least would you mind renaming your variables correctly please? You call them "ref" which implies they are React references but it's not what they are, they are HTML elements (you got the types right though). So please rename it to "newNoteButtonElement" for example.